src/App.tsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
-
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [error, setError] = useState<null | string>(null);
+
+  const handleClick = () => getAll()
+    .then((receivedGoods) => setGoods(receivedGoods));
+
+  const handle5FirstClick = () => get5First()
+    .then((receivedGoods) => setGoods(receivedGoods));
+
+  const handleRedClick = () => getRedGoods()
+    .then((receivedGoods) => setGoods(receivedGoods))
+    .catch(() => setError('Something went wrong'));
```

Comment:
set the error for all of the functions

Reply:
also think about the place to clear the error so that it doesnt persist on the website